pandas/tests/frame/test_constructors.py

```diff
@@ -312,8 +312,9 @@ def test_constructor_dtype_nocast_view_2d_array(
         df = DataFrame([[1, 2], [3, 4]], dtype="int64")
         if not using_array_manager and not using_copy_on_write:
             should_be_view = DataFrame(df.values, dtype=df[0].dtype)
```

Comment:
Why should this warn? Not a blocking comment

Reply:
I was thinking because currently it is a view and changes the parent dataframe? 
But the input here is a numpy array, so not sure if we know that / if we have a way to warn users about the change in behaviour (essentially copy=False becomes copy=True if the input is a 2D ndarray)

Reply:
No there is no way of knowing where the NumPy array came from, we have no clue that It was the backing array of a DataFrame before that

Reply:
I assume that's something we will simply have to document in the list of breaking changes.